packages/ui-scripts/src/publish.ts

```diff
@@ -22,34 +22,37 @@
  * SOFTWARE.
  */
 import { getPackageJSON, getPackages } from '@instructure/pkg-utils'
-import {
-  error,
-  info,
-  runCommandSync,
-  runCommandAsync
-} from '@instructure/command-utils'
+import { error, info, runCommandAsync } from '@instructure/command-utils'
 
 import { getConfig } from './utils/config'
-import { checkWorkingDirectory, isReleaseCommit, setupGit } from './utils/git'
-import { publishPackages, createNPMRCFile } from './utils/npm'
-
-try {
-  const pkgJSON = getPackageJSON(undefined)
-  // ui-scripts --publish               - to publish from the master branch
-  // ui-scripts --publish maintenance   - to publish from any legacy branch
-  const isMaintenance = process.argv[3] === 'maintenance'
-
-  publish({
-    packageName: pkgJSON.name,
-    version: pkgJSON.version,
-    isMaintenance,
-    //TODO investigate if config is needed
-    config: getConfig(pkgJSON)
-  })
-} catch (err: any) {
-  error(err)
-  process.exit(1)
-}
+import {
+  checkWorkingDirectory,
+  isReleaseCommit,
+  runGitCommand,
+  setupGit
+} from './utils/git'
+import { bumpPackages, createNPMRCFile } from './utils/npm'
+import semver from 'semver'
+
+type PackageInfo = ReturnType<typeof getPackages>[number]
+;(async function doPublish() {
+  try {
+    const pkgJSON = getPackageJSON()
+    // ui-scripts --publish               - to publish from the master branch
+    // ui-scripts --publish maintenance   - to publish from any legacy branch
+    const isMaintenance = process.argv[3] === 'maintenance'
+
+    await publish({
+      packageName: pkgJSON.name!,
+      version: pkgJSON.version!,
+      isMaintenance,
+      config: getConfig(pkgJSON)
+    })
+  } catch (err: any) {
+    error(err)
+    process.exit(1)
+  }
+})()
```

Comment:
The previous version had nested try-catch blocks and some of the catch blocks were simply swallowing the error, this new way only has 1 try-catch which means any error we might encounter during the release process will halt the whole process.